### ISSUE TITLE
Select/Dropdowns don't meet a11y contrast ratio

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/design-gc-conception#2086]
Location: Submit a Holiday
Date Tested: 2025-12-05
Status: Fail
Severity: Serious
WCAG Criterion: 1.4.03 Contrast (Minimum) (Level A)

Description:
Text fails to meet contrast minimums under WCAG guidelines.

Example 1: “Select a month” option highlighted in #C8C8C8 vs white text #FFFFFF (1.67:1).

Example 2: In High Contrast Mode, the Government of Canada logo becomes difficult to read.

Recommendation:
Increase foreground/background contrast for all text to ≥4.5:1 (normal) and ≥3:1 (large).

<img width="985" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL0ZkREE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5232932615cada7ff5101c96bc745ef0de69186e/image.png" alt="image.png" />
<img width="893" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL0pkREE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4c97d415232f23be3fd928e43bdb41ceaa6b9878/image.png" alt="image.png" />

**Comment by melaniebmn** [comment]
Hey, added the "Design" label and added "Design" to the ticket heading since this needs a design review first before we can change it in the code. As Ethan said, we finally have the ability to add custom select styling with CSS. Here is a blog post from Chrome for developers that shows some examples of what is possible:

https://developer.chrome.com/blog/a-customizable-select

**Comment by melaniebmn** [comment]
@adorayi should we create a duplicate ticket for dev once the design has been reviewed / done?

**Comment by nmakuch** [comment]
Hi all, I had started some work on this in Figma but as I've done more research and reading I'm realizing we don't need to make a change here.

As we know these are browser defaults which we don't have control over, but they do meet a11y standards.

In terms of the first screenshot, the fill is not the only cue here it's working in combination with the cursor switching to a pointer, which is a pretty common pattern and even shows up in some of the most accessible design systems two I've noticed are: 
- Atlassian in their side nav and in their select component (https://atlassian.design/components/select/examples) 
- France in their side menu (https://www.systeme-de-design.gouv.fr/version-courante/fr)

In terms of the second screen shot, the styling highlighted is actually on a disabled input and can be considered omitted from the contrast check. I did a test on my tablet using Edge on the other options and the colours meet contrast standards (see screenshot). 

<img width="656" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMkVXRFE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b90ec2946b98a85fc9741b81b2aa2c5f910d604f/Screenshot%202026-02-27%20153419.png" alt="Screenshot 2026-02-27 153419.png" />

The colours they are using for the active elements in the dropdown are using a background of 0x767676 and text is 0xFFFFFF (contrast ratio of **4.54:1** AA)

So for the documented reasons above I am suggesting to not make any changes here

**Comment by nmakuch** [comment]
It does make sense to work on a custom dropdown if we ever get around to designing / developing an auto-complete component, but until then I think it's best to stick with the native browser ui
